### PR TITLE
creating math_constants.h file - attempt #2

### PR DESCRIPTION
**Pull request by deven-amd**
this PR is really a continuation of PR #512

### CODE REVIEW

**Comment by gargrahul** [comment]
I don't think this is the right way. On HIP/NVCC we should still be using existing cuda/.../math_constants.h.

**Comment by gargrahul** [comment]
I took an existing unit test for modification, which passes always too.

https://github.com/RadeonOpenCompute/hcc/blob/2e7519287cc90b02fd4049eeae2d50fba014ac83/tests/Unit/AM/am_memtracker.cpp

This test should always pass since allocated memories should get freed correctly always.

If you run this test without my fix, you will see one tracker entry despite 2 allocations, that was the problem.

**Comment by deven-amd** [comment]
@gargrahul, I do not think I follow you on your suggestion about using the CUDA version of math_constants.h for the hip/nvcc path.

Note that the #defines have `HIPRT` prefix on them, and that is what is referenced in the (user) source code. Given that how can we use the CUDA version for math_constants.h for the hip/nvcc path?

File: include/hip/math_constants.h [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+#ifndef HIP_INCLUDE_HIP_MATH_CONSTANTS_H
+#define HIP_INCLUDE_HIP_MATH_CONSTANTS_H
+
+
+// Create architecture (hcc / nvcc) specific headers if/when a constant value
+// needs to be defined differently for either of the two
+
+/* single precision constants */
+
+#define HIPRT_INF_F        __int_as_float(0x7f800000)
```
[the remark this answers is not in the document]
Comment (AlexVlx) [reply include/hip/math_constants.h]: That's not the macro you want, you want the one in cmath / math.h which is `HUGE_VALF` and it's specified to expand into a `float` constant, how that is implemented is not the HIPRT's business (and `__HUGE_VALF_bytes` may or may not be an implementation detail for your standard library).  Which brings us to the question of which macros are missing an equivalent? AFAICS:
- `HIPRT_NAN_F`                 -> `NAN` (https://en.cppreference.com/w/cpp/numeric/math/NAN)
- `HIPRT_MIN_DENORM_F` -> `FLT_TRUE_MIN` (https://en.cppreference.com/w/cpp/types/climits)
- `HIPRT_MAX_NORMAL_F` -> `FLT_MAX` (https://en.cppreference.com/w/cpp/types/climits)
- `HIPRT_NEG_ZERO_F`      -> `-0.0f`
- `HIPRT_ZERO_F`                -> `0.0f`
- `HIPRT_ONE_F`                  -> `1.0f`
Are there some others which are not part of the PR / that I skipped?

**Comment by gargrahul** [comment]
@deven-amd In HIP applications we will continue to use #defines with HIPRT prefix however on HIP/NVCC these #defines will map to CUDA versions of #defines which are part of existing cuda/math_constants.h

I can do those changes once you and @AlexVlx confirm that we need these as discussed in other comment.

**Comment by AlexVlx** [comment]
@gargrahul I'm not saying we do not need these, just that we should use standard features where available.

**Comment by deven-amd** [comment]
I am going to try out @AlexVlx  in the Eigen codebase (where these constants are used) and once done will provide an update as to whether that suggestion works or not.  Expect the update by end-of-tomorrow.

We can close/merge the PR based on the outcome.

**Comment by mangupta** [comment]
@deven-amd Is there any plans to update this PR or can I go ahead and close this?

**Comment by mangupta** [comment]
Closing this since there is no update from author on this PR.

**Pull request closed without merging**